pkg/cli/admin/ocpcertificates/certgraphanalysis/certkeypair.go

```diff
@@ -0,0 +1,213 @@
+			CertMetadata:    toCertKeyMetadata(certificate),
+		},
+	}
+
+	details, err := toCertKeyPairDetails(certificate)
+	ret.Spec.Details = details
+	if err != nil {
+		ret.Status.Errors = append(ret.Status.Errors, err.Error())
+	}
+
+	return ret, nil
+}
+
+func toCertKeyPairDetails(certificate *x509.Certificate) (certgraphapi.CertKeyPairDetails, error) {
+	count := 0
+	isClient := false
+	isServing := false
+	isSigner := (certificate.KeyUsage & x509.KeyUsageCertSign) != 0
+	if isSigner {
+		count++
+	}
+	for _, curr := range certificate.ExtKeyUsage {
+		if curr == x509.ExtKeyUsageClientAuth {
+			isClient = true
+			count++
+		}
+		if curr == x509.ExtKeyUsageServerAuth {
+			isServing = true
+			count++
+		}
+	}
+	var typeError error
+	if count == 0 {
+		typeError = fmt.Errorf("you have a cert for nothing?")
+	}
+	if count > 1 {
+		typeError = fmt.Errorf("you have a cert for more than one?  We don't do that. :(")
+	}
+
+	ret := certgraphapi.CertKeyPairDetails{}
+	if isClient {
+		ret.CertType = "ClientCertDetails"
+		ret.ClientCertDetails = toClientCertDetails(certificate)
+	}
+	if isServing {
+		ret.CertType = "ServingCertDetails"
+		ret.ServingCertDetails = toServingCertDetails(certificate)
+	}
+	if isSigner {
+		ret.CertType = "SignerCertDetails"
+		ret.SignerDetails = toSignerDetails(certificate)
+	}
+
+	if count > 1 {
+		ret.CertType = "Multiple"
+	}
+
+	return ret, typeError
+}
+
+func toClientCertDetails(certificate *x509.Certificate) *certgraphapi.ClientCertDetails {
+	return &certgraphapi.ClientCertDetails{
+		Organizations: certificate.Subject.Organization,
+	}
+}
+
+func toServingCertDetails(certificate *x509.Certificate) *certgraphapi.ServingCertDetails {
+	ret := &certgraphapi.ServingCertDetails{
+		DNSNames:    certificate.DNSNames,
+		IPAddresses: nil,
+	}
+
+	for _, curr := range certificate.IPAddresses {
+		ret.IPAddresses = append(ret.IPAddresses, curr.String())
+	}
+
+	return ret
+}
+
+func toSignerDetails(certificate *x509.Certificate) *certgraphapi.SignerCertDetails {
+	return &certgraphapi.SignerCertDetails{}
+}
+
+func toCertKeyMetadata(certificate *x509.Certificate) certgraphapi.CertKeyMetadata {
+	ret := certgraphapi.CertKeyMetadata{
+		CertIdentifier: certgraphapi.CertIdentifier{
+			CommonName:   certificate.Subject.CommonName,
+			SerialNumber: certificate.SerialNumber.String(),
+		},
+		SignatureAlgorithm: certificate.SignatureAlgorithm.String(),
+		PublicKeyAlgorithm: certificate.PublicKeyAlgorithm.String(),
+		ValidityDuration:   duration.HumanDuration(certificate.NotAfter.Sub(certificate.NotBefore)),
+	}
+
+	switch publicKey := certificate.PublicKey.(type) {
+	case *ecdsa.PublicKey:
+		ret.PublicKeyBitSize = fmt.Sprintf("%d bit, %v curve", publicKey.Params().BitSize, publicKey.Params().Name)
+	case *rsa.PublicKey:
+		ret.PublicKeyBitSize = fmt.Sprintf("%d bit", publicKey.Size()*8)
+	default:
+		fmt.Fprintf(os.Stderr, "%T\n", publicKey)
+	}
+
+	signerHumanName := certificate.Issuer.CommonName
```

Comment:
use the full `Issuer.String()`

Reply:
If memory serves, that did bad things in the generated doc, but try it in cluster-debug-tools against a cluster and we can tweak here if appropriate.